CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -618,6 +618,7 @@ function UIEditRoom:returnToDoorPhase()
         if not obj.master then
           self:addObjects({{
             object = TheApp.objects[obj.object_type.id],
+            state = obj:getState(),
```

Comment:
You might want to pull the state before the object is destroyed, in case the destruction breaks the state?